### PR TITLE
[Performance] Recreate schema-native performance refactor

### PR DESCRIPTION
**Pull request by PragalvaXFREZ**
## Description

Recreates the schema-native performance refactor from meshery/meshery#19605 after its revert, while rebasing the work onto current master.

This switches performance test/profile handling from the legacy service-mesh-performance construct to Meshery Schemas performance profile types, keeps compatibility route handling, and moves the performance UI toward schema-generated API hooks.

The root module is pinned to `github.com/meshery/schemas v1.3.6` so the recreated code can consume the current generated performance profile types. The unrelated wasm module, UI lockfile, Docker, and workflow churn from the earlier reverted branch is not included.

## Validation

- `go test ./server/models ./server/handlers ./mesheryctl/internal/cli/root/perf`
- `npm test -- --run rtk-query/__tests__/performance-profile.test.ts`
- `git diff --check HEAD~1..HEAD`

### CODE REVIEW

File: ui/components/performance/Dashboard.tsx [hunk 1]
```diff
@@ -64,74 +64,60 @@ function Dashboard() {
...
 
-  /**
-   * fetch performance profiles when the page loads
-   */
   useEffect(() => {
-    fetchTestProfiles();
-    fetchTests();
-  }, []);
+    updateProgress({ showProgress: isFetchingProfiles || isFetchingResults });
+  }, [isFetchingProfiles, isFetchingResults]);
```
[the remark this answers is not in the document]
Comment (PragalvaXFREZ) [reply ui/components/performance/Dashboard.tsx]: Addressed in `afda0d3594e4`. I used `useDispatch()` with `updateProgressAction(...)` here instead of wrapping `updateProgress(...)` directly because `updateProgress` is a helper that already dispatches through the global store. Using the actual slice action keeps this component on the normal Redux hook path and avoids double-dispatching a helper.

File: ui/components/performance/PerformanceCalendar.tsx [hunk 2]
```diff
@@ -96,42 +96,34 @@ function PerformanceCalendar({ style }) {
...
-            // @ts-ignore
-            setResults(result.results || []);
-          }
-        }
-      },
-      error: handleError('Failed to Fetch Profiles'),
-    });
-  }
+    updateProgress({ showProgress: isFetching });
+  }, [isFetching]);
```
[the remark this answers is not in the document]
Comment (PragalvaXFREZ) [reply ui/components/performance/PerformanceCalendar.tsx]: Addressed in `afda0d3594e4`. I changed this component to use `useDispatch()` with `updateProgressAction(...)`. I intentionally did not use `dispatch(updateProgress(...))` because `updateProgress` is a global-store helper, while `updateProgressAction` is the actual Redux action creator.

File: ui/components/performance/PerformanceProfiles.tsx [hunk 3]
```diff
@@ -59,84 +64,47 @@ function PerformanceProfile({ handleDelete }) {
...
-          order: `${sortOrder}`,
-        },
-      },
-    );
-    return () => {
-      subscription.dispose();
-    };
-  }, [page, pageSize, search, sortOrder]);
+    updateProgress({ showProgress: isFetchingProfiles });
+  }, [isFetchingProfiles]);
```
[the remark this answers is not in the document]
Comment (PragalvaXFREZ) [reply ui/components/performance/PerformanceProfiles.tsx]: Addressed in `afda0d3594e4`. This now dispatches `updateProgressAction(...)` through the component dispatch hook. I also updated the related success/error progress updates in this component so the progress state uses one consistent Redux path.

File: ui/components/performance/PerformanceResults.tsx [hunk 4]
```diff
@@ -501,42 +507,41 @@ function MesheryResults({ endpoint, CustomHeader = <div />, elevation = 4 }) {
...
+    },
+    {
+      skip: !profileID,
+    },
+  );
 
   useEffect(() => {
-    fetchResults(page, pageSize, search, sortOrder);
+    updateProgress({ showProgress: isFetching });
+  }, [isFetching]);
```
[the remark this answers is not in the document]
Comment (PragalvaXFREZ) [reply ui/components/performance/PerformanceResults.tsx]: Addressed in `afda0d3594e4`. `dispatch` already existed here, so I switched the progress updates to dispatch `updateProgressAction(...)` directly. That preserves the intended Redux update while avoiding the global `updateProgress` helper in this component.

File: mesheryctl/internal/cli/root/perf/apply.go [hunk 5]
```diff
@@ -58,6 +58,16 @@ var linkDocPerfApply = map[string]string{
...
 
+func splitTestDuration(duration string) (string, string) {
+	duration = strings.TrimSpace(duration)
+	if duration == "" {
+		duration = "30s"
+	}
+
+	durLen := len(duration)
+	return duration[:durLen-1], strings.ToLower(duration[durLen-1:])
+}
```
[the remark this answers is not in the document]
Comment (PragalvaXFREZ) [reply mesheryctl/internal/cli/root/perf/apply.go:78]: Addressed in `afda0d3594e4`. I changed `splitTestDuration` to find the first non-numeric/non-decimal character, default unitless durations to seconds, and preserve multi-character units like `ms`. I also added tests for the two edge cases called out here: `30` and `500ms`.

File: server/models/default_local_provider.go [hunk 6]
```diff
@@ -1025,24 +1025,28 @@ func (l *DefaultLocalProvider) ShareFilter(_ *http.Request) (int, error) {
...
+	}
+
+	if performanceProfile.ID == uuid.Nil {
+		uid, err := uuid.NewV4()
 		if err != nil {
 			return nil, ErrGenerateUUID(err)
 		}
-		performanceProfile.ID = &uid
+		performanceProfile.ID = uid
+	}
```
[the remark this answers is not in the document]
Comment (PragalvaXFREZ) [reply server/models/default_local_provider.go]: Addressed in `afda0d3594e4`. `core.Uuid` is currently an alias to `uuid.UUID`, but this code is operating at the schema/model boundary, so I changed the nil check to compare against `core.Uuid{}`. The generated ID assignment remains unchanged because `uuid.NewV4()` already returns the compatible underlying UUID value.

**Comment by PragalvaXFREZ** [comment]
Added backward compatibility for the schema-native performance profile refactor. Existing saved performance test configs can now be read from both the new schema JSON shape and the older SMP JSON shape, including legacy snake_case fields and numeric protocol values. Profile reads now tolerate older local rows with missing user_id values, and new profile saves stamp the authenticated user ID so local saves do not write a zero userId. The list path now returns a decode error instead of silently dropping unreadable saved configs. I also added focused tests for the legacy and schema-native performance test config decode paths.

**Review by hortison — APPROVED** [review]
lgtm

**Comment by marblom007** [comment]
looks good.
